surveys/2021-annual-survey/questions.md

```diff
@@ -123,9 +123,33 @@ Select all that apply:
 
 ## Your Rust experience
 
+### When did you learn to program in Rust?
+
+**Note**: while you may continue to try to improve your Rust skills, for this question assume "learning to program in Rust" means spending the majority of your time with Rust consuming learning materials or coding *in order to learn* (as opposed to achieving some other goal).
+
+Select one:
+
+- I'm still *actively* trying to learn Rust
+- Within the last year
+- Between 1 to 3 years ago
+- Between 3 to 6 years ago
+- More than 6 years ago
+
+> **justification**
+>
+> Useful for cohort analysis, i.e., for other questions we can understand if *when* someone learned Rust impacts their views on things.
+>
+> The time periods used as answers try to reflect the major "epochs" of Rust history (i.e., pre-1.0, 2015 edition, 2018 edition, etc.).
```

Comment:
It might be easier to phrase it like this? I.e., what was the first version of Rust you used? Pre-1.0, 2015, 2018, currently learning, or something like that. Or maybe keep the years and ask "when did you start learning Rust"

Reply:
I think it is an improvement, but I would still prefer to ask when people started learning Rust, since that is a point in time, rather than a period and so makes it less confusing to match with answers (and also removes the potential ambiguity around when a person is done learning)

Reply:
I get the concern. Ideally we would want to know what period of time was most formative for their learning as it's not really important when they started or ended learning in their eyes but what was Rust like when they were deepest in "learning mode". This is obviously not really possible to ask, so I'm trying to ask it in a way that feels closest to that.

No matter which way we ask there will always be people who are unsure if which category they fall into.